src/main/java/org/jenkinsci/plugins/configfiles/ConfigContextResolver.java

```diff
@@ -0,0 +1,26 @@
+package org.jenkinsci.plugins.configfiles;
+
+import hudson.ExtensionList;
+import hudson.ExtensionPoint;
+import hudson.model.ItemGroup;
+import jenkins.model.Jenkins;
+
+import java.util.Optional;
+
+/**
+ * ConfigContextResolver provides a way for a plugin to specify the context ({@link ItemGroup}) used to retrieve
+ * configuration files from for classes deriving from {@link ItemGroup} it defines.
+ */
+public abstract class ConfigContextResolver implements ExtensionPoint {
+
+    public static ExtensionList<ConfigContextResolver> all() {
+        return Jenkins.getInstance().getExtensionList(ConfigContextResolver.class);
+    }
+    /**
+     * Optionally provides the {@link ItemGroup} from which configuration files should be retrieved for the provided
+     * one.

```

Comment:
I think for other developers it would be helpful if you could add your origin usecase with the `promoted-builds-plugin`